metricbeat/module/mongodb/status/status.go

```diff
@@ -62,20 +57,34 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // Fetch methods implements the data gathering and data conversion to the right format
 // It returns the event which is then forward to the output. In case of an error, a
 // descriptive error must be returned.
-func (m *MetricSet) Fetch() (common.MapStr, error) {
+func (m *MetricSet) Fetch(r mb.ReporterV2) {
 
 	// instantiate direct connections to each of the configured Mongo hosts
 	mongoSession, err := mongodb.NewDirectSession(m.DialInfo)
 	if err != nil {
-		return nil, err
+		r.Error(err)
+		return
 	}
 	defer mongoSession.Close()
 
 	result := map[string]interface{}{}
 	if err := mongoSession.DB("admin").Run(bson.D{{Name: "serverStatus", Value: 1}}, &result); err != nil {
-		return nil, err
+		r.Error(err)
+		return
 	}
 
-	data, _ := schema.Apply(result)
-	return data, nil
+	event := mb.Event{}
+	event.MetricSetFields, _ = schema.Apply(result)
+
+	if v, err := event.MetricSetFields.GetValue("version"); err == nil {
+		event.RootFields = common.MapStr{}
+		event.RootFields.Put("service.version", v)
+		event.MetricSetFields.Delete("version")
+	}
+	if v, err := event.MetricSetFields.GetValue("process"); err == nil {
+		event.RootFields = common.MapStr{}
```

Comment:
Can both `version` and `process` be present? If they are `RootFields` will get overwritten here.

Reply:
yes, good find. Will fix it.